Signal/src/ViewControllers/LoadingViewController.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import PromiseKit
+
+// The initial presentation is intended to be indistinguishable from the Launch Screen.
+// After a delay we present some "loading" UI so the user doesn't think the app is frozen.
+@objc
+public class LoadingViewController: UIViewController {
+
+    var logoView: UIImageView!
+    var topLabel: UILabel!
+    var bottomLabel: UILabel!
+
+    override public func loadView() {
+        self.view = UIView()
+        view.backgroundColor = UIColor.ows_materialBlue
+
+        self.logoView = UIImageView(image: #imageLiteral(resourceName: "logoSignal"))
+        view.addSubview(logoView)
+
+        logoView.autoCenterInSuperview()
+        logoView.autoPinToSquareAspectRatio()
+        logoView.autoMatch(.width, to: .width, of: view, withMultiplier: 1/3)
+
+        self.topLabel = buildLabel()
+        topLabel.alpha = 0
+        topLabel.font = UIFont.ows_dynamicTypeTitle2
+        topLabel.text = NSLocalizedString("DATABASE_VIEW_OVERLAY_TITLE", comment: "Title shown while the app is updating its database.")
+
+        self.bottomLabel = buildLabel()
+        bottomLabel.alpha = 0
+        bottomLabel.font = UIFont.ows_dynamicTypeBody
+        bottomLabel.text = NSLocalizedString("DATABASE_VIEW_OVERLAY_SUBTITLE", comment: "Subtitle shown while the app is updating its database.")
+
+        let labelStack = UIStackView(arrangedSubviews: [topLabel, bottomLabel])
+        labelStack.axis = .vertical
+        labelStack.alignment = .center
+        labelStack.spacing = 8
+        view.addSubview(labelStack)
+
+        labelStack.autoPinEdge(.top, to: .bottom, of: logoView, withOffset: 20)
+        labelStack.autoPinLeadingToSuperviewMargin()
+        labelStack.autoPinTrailingToSuperviewMargin()
+        labelStack.setCompressionResistanceHigh()
+        labelStack.setContentHuggingHigh()
```

Comment:
We should talk about layout sometime.  I've stopped using the "low" flavors of these methods since they can have unpredictable results.  So I use just "default" and "high".

Reply:
In practice I rarely use `low`, but I certainly haven't avoided it, and can imagine certain circumstances where I *would* use it. e.g. if I have ten objects in a stack view, 9 of which I want to stay small and only one which I want to grow, I'd apply low content hugging to the one view, rather than high hugging to the other 9.

I'd be glad to know what these unpredictable results are about.